src/sqllogictest/src/runner.rs

```diff
@@ -506,7 +506,7 @@ impl Runner {
             threads: NUM_TIMELY_WORKERS,
             process: 0,
             addresses: vec![SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), 0)],
-            data_directory: None,
+            data_directory: tempfile::tempdir().unwrap().into_path(),
```

Comment:
You can return the error here instead of panicking (`tempfile::tempdir()?`), but also this should do the same trick as mentioned in the last comment to keep the temporary directory alive as long as `Runner`!